src/simple-keyring.ts

```diff
@@ -126,11 +127,19 @@ export default class SimpleKeyring implements Keyring<string[]> {
     opts: KeyringOpt = { version: SignTypedDataVersion.V1 },
   ) {
     // Treat invalid versions as "V1"
-    const version = Object.keys(SignTypedDataVersion).includes(
-      opts.version || '',
-    )
-      ? (opts.version as SignTypedDataVersion)
-      : SignTypedDataVersion.V1;
+    let version = SignTypedDataVersion.V1;
+
+    // Type predicate type guard to check if a string is in the enum SignTypedDataVersion
+    // TODO: Put this in @metamask/eth-sig-util
+    const isSignTypedDataVersion = (
```

Comment:
Nit: Might be better to place this outside of `signTypedData` so it doesn't get redefined every time.